public/components/data_connections/components/access_control_tab.tsx

```diff
@@ -0,0 +1,205 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import {
+  EuiButton,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiSpacer,
+  EuiText,
+  EuiHorizontalRule,
+  EuiBottomBar,
+  EuiButtonEmpty,
+} from '@elastic/eui';
+import React, { useEffect, useState } from 'react';
+import { EuiPanel } from '@elastic/eui';
+import {
+  ACCELERATION_ALL,
+  ACCELERATION_RESTRICT,
+  QUERY_ALL,
+  QUERY_RESTRICT,
+} from 'common/constants/data_connections';
+import { AccessControlCallout } from './access_control_callout';
+import { coreRefs } from '../../../../public/framework/core_refs';
+import { QueryPermissionsFlexItem } from './query_permissions_flex_item';
+import { AccelerationPermissionsFlexItem } from './acceleration_permissions_flex_item';
+
+export const AccessControlTab = () => {
+  const [mode, setMode] = useState<'view' | 'edit'>('view');
+  const [roles, setRoles] = useState<Array<{ label: string }>>([]);
+  const [selectedQueryPermissionRoles, setSelectedQueryPermissionRoles] = useState<
+    Array<{ label: string }>
+  >([]);
+  const [selectedAccelerationPermissionRoles, setSelectedAccelerationPermissionRoles] = useState<
+    Array<{ label: string }>
+  >([]);
+  const [queryPermissionRadioSelected, setQueryRadioIdSelected] = useState(`query-1`);
+  const [accelerationPermissionRadioSelected, setAccelerationRadioIdSelected] = useState(
+    `acceleration-1`
+  );
+
+  const [finalQueryPermissionsPlaceHolder, setFinalQueryPermissionsPlaceHolder] = useState<
+    Array<{ label: string }>
+  >([]);
+  const [
+    finalAccelerationPermissionsPlaceHolder,
+    setFinalAccelerationPermissionsPlaceHolder,
+  ] = useState<Array<{ label: string }>>([]);
+
+  useEffect(() => {
+    coreRefs.http!.get('/api/v1/configuration/roles').then((data) =>
+      setRoles(
+        Object.keys(data.data).map((key) => {
+          return { label: key };
+        })
+      )
+    );
```

Comment:
Assumes security plugin is installed and user has permission to access it (will be addressed with the permission PR)

Reply:
What happens if security plugin is not installed. Are we not showing the whole data connections plugin? or Do we have a fallback view?

Reply:
I think current plan is to show as much as possible:
- if users have access to datasources APIs, we show them the page, and if they do not then we will show Fallback UI
- if users have access to datasource APIs, but do not have access to security plugin or security plugin is not installed, we will not be able to show the dropdown with pre-defined roles, but we will still allow them to manually type in the roles. Created a follow up issue to avoid scope-creep: https://github.com/opensearch-project/dashboards-observability/issues/1022. Will not implement manually type to create roles in this PR, but in follow up PR.